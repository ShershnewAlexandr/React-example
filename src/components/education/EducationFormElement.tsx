import React from 'react';
import { Grid } from '@material-ui/core';
import { Field } from 'redux-form/lib/immutable';
import { WithTranslation } from 'react-i18next';
import TextInput from '../inputs/TextInput';
import DateInput from '../inputs/DateInput';
import EndDateInputContainer from '../../containers/EndDateInputContainer';

interface EducationFormElementProps {
  element: string;
  disabled: boolean;
}

function EducationFormElement({
  element,
  disabled,
  t,
}: EducationFormElementProps & WithTranslation) {
  return (
    <Grid
      container
      direction="row"
      spacing={4}
      data-testid="education-form-block"
    >
      <Grid item xs={12}>
        <Field
          name={`${element}.schoolName`}
          component={TextInput}
          label={t('education.schoolname')}
          disabled={disabled}
        />
      </Grid>
      <Grid item xs={6}>
        <Field
          name={`${element}.startYear`}
          component={DateInput}
          label={t('education.startYear')}
          disabled={disabled}
        />
      </Grid>
      <Grid item xs={6}>
        <Field
          name={`${element}.endYear`}
          component={EndDateInputContainer}
          label={t('education.endYear')}
          disabled={disabled}
        />
      </Grid>
    </Grid>
  );
}

export default EducationFormElement;
