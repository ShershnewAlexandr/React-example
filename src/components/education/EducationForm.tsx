import React from 'react';
import { FieldArray } from 'redux-form/lib/immutable';
import EducationFormElement from './EducationFormElement';
import DropAddListForm from '../dropAddList/DropAddListForm';
import { EducationFormContainerProps } from '../../containers/education/EducationFormContainer';

function EducationForm({
  handleSubmit,
  isLoading,
}: EducationFormContainerProps) {
  return (
    <form onSubmit={handleSubmit} data-testid="educationForm">
      <FieldArray
        name="education"
        component={DropAddListForm}
        formRender={EducationFormElement}
        maxElements={5}
        disabled={isLoading}
      />
    </form>
  );
}

export default EducationForm;
