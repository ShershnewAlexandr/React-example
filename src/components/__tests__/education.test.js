import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import EducationForm from '../education/EducationForm';
import EducationFormElement from '../education/EducationFormElement';

describe('educationForm', () => {
  const propsForm = {
    handleSubmit: jest.fn(),
    isLoading: false,
  };
  const propsFormElement = {
    element: null,
    disabled: false,
    t: jest.fn(),
  };

  const educationForm = shallow(<EducationForm {...propsForm} />);
  const educationFormElement = shallow(
    <EducationFormElement {...propsFormElement} />,
  );

  it('snapshot Form', () => {
    expect(toJson(educationForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    educationForm.find('form').simulate('submit');
    expect(propsForm.handleSubmit.mock.calls.length).toBe(1);
  });

  it('snapshot FormElement', () => {
    expect(toJson(educationFormElement)).toMatchSnapshot();
  });
});
