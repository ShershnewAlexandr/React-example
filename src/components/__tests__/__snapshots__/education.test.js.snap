// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`educationForm snapshot Form 1`] = `
<form
  data-testid="educationForm"
  onSubmit={[MockFunction]}
>
  <FieldArray
    component={[Function]}
    disabled={false}
    formRender={[Function]}
    maxElements={5}
    name="education"
  />
</form>
`;

exports[`educationForm snapshot FormElement 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-testid="education-form-block"
  direction="row"
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.schoolName"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.startYear"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.endYear"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
